kernel/src/actions/mod.rs

```diff
@@ -152,6 +199,81 @@ impl Protocol {
             .as_ref()
             .is_some_and(|features| features.iter().any(|f| f == feature.as_ref()))
     }
+
+    /// Check if reading a table with this protocol is supported. That is: does the kernel support
+    /// the specified protocol reader version and all enabled reader features? If yes, returns unit
+    /// type, otherwise will return an error.
+    pub fn check_read_supported(&self) -> DeltaResult<()> {
```

Comment:
nit: when I'm returning a `Result<()>` like this I prefer `ensure_read_supported`, since "check" implies (to me at least) that it should just check and return true/false.

Same for `check_write_supported` below

Reply:
cool sgtm :)